soci/soci_index.go

```diff
@@ -365,8 +387,9 @@ func (b *IndexBuilder) buildSociLayer(ctx context.Context, desc ocispec.Descript
 	}
 
 	if !b.ztocBuilder.CheckCompressionAlgorithm(compressionAlgo) {
-		return nil, fmt.Errorf("layer %s (%s) is compressed in an unsupported format. expect: [tar, gzip, unknown] but got %q: %w",
-			desc.Digest, desc.MediaType, compressionAlgo, errUnsupportedLayerFormat)
+		fmt.Printf("ztoc skipped - layer %s (%s) is compressed in an unsupported format. expect: [tar, gzip, unknown] but got %q",
```

Comment:
Now that an unsupported compression algorithm causes a skip instead of an error, I think it would make sense to move this whole check into `skipBuildingZtoc` and return this a reason rather than logging it directly.

also nit: This is missing `\n`

Reply:
IMO it's fine to keep it as it is. We can open another PR if we believe this logic gets more complicated and needs refactor in the future. This PR does not need to be blocked because of it.